### PR TITLE
feat: add a static copy of legacy proto-based types

### LINKED ISSUE #999: Keep legacy generated SQL types available for import
As a follow-up to #814 and as discussed with @tswast on the chat, some users, that otherwise could upgrade to a `3.x` version, would still need the legacy generated SQL model types that the `3.x` version will remove.

We should add these models back to the codebase, but with the following restrictions:
- These legacy files would be copy-pasted to their old place, but will not be generated anymore.
- The legacy models would not be exposed in the library's top-level namespace, nor would they be imported or used anywhere else in the code. Users will have to import them using the full module path, and at their own risk.
- These types will not be maintained in any way, and will eventually get out of sync with what the code generator would produce. This is intentional.
- When importing anything from that legacy sub-package, a warning should be issued.

@tswast Feel free to update/correct the requirements above in case I misunderstood/misremembered them.

### PR DESCRIPTION
**Pull request by plamut**
Closes #999.

Adding unmaintained legacy types definitions.

- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: google/cloud/bigquery_v2/__init__.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+from .types.standard_sql import StandardSqlTableType
+from .types.table_reference import TableReference
+
+
+_LEGACY_MSG = (
+    "Legacy proto-based types from bigquery_v2 are not maintained anymore, "
+    "use types defined in google.cloud.bigquery instead."
+)
+
+warnings.warn(_LEGACY_MSG, category=DeprecationWarning)
```
Comment (tseaver) [inline google/cloud/bigquery_v2/__init__.py:39]: Nice -- I had been thinking to add a request for import-time warnings.

**Review by tseaver — APPROVED** [review]

File: google/cloud/bigquery_v2/__init__.py (hunk 1 above)
Comment (plamut) [reply google/cloud/bigquery_v2/__init__.py:39]: Thanks!

**Comment by plamut** [comment]
Oh, wait... didn't we say we want to add back the proto* dependencies, so that those imports will not result in an error? Maybe as a "legacy" extra?

And what about the intersphinx links? Should we perhaps expose these models in the docs with a big fat "OBSOLETE!" warning at the top?

@tswast

**Comment by tswast** [comment]
> Oh, wait... didn't we say we want to add back the proto* dependencies, so that those imports will not result in an error? Maybe as a "legacy_types" extra?

Yes, please. I think we've added the BQ Storage API as a required dependency now, right? That should mean we pull those in transitively anyway, but it wouldn't hurt to be explicit.

> And what about the intersphinx links? Should we perhaps expose these models in the docs with a big fat "OBSOLETE!" warning at the top?

Yeah, let's make sure we generate docs for it, but with a warning & link to our migration guide.

**Review by tswast — APPROVED** [review]

**Pull request merged**